b+tree.ts

```diff
@@ -488,6 +488,213 @@ export default class BTree<K=any, V=any> implements ISortedMapF<K,V>, ISortedMap
     return {nodequeue, nodeindex, leaf:nextnode};
   }
 
+  /**
+   * Computes the delta operations (adds, deletes, and modifications) required to go from this tree to the supplied destination.
+   * For efficiency, the delta is returned via invocations of supplied handlers.
+   * The computation is optimized for the case in which the two trees have large amounts of shared data (obtained by calling the
+   * `clone` or `with` APIs) and will avoid any iteration of shared state.
+   * @param destination The tree to compute a delta to.
+   * @param processAdd Callback invoked for all adds in the delta.
+   * @param processDelete Callback invoked for all deletions in the delta.
+   * @param processModify Callback invoked for all modifications in the delta.
+   */
+  delta(destination: BTree<K, V>, processAdd: (k: K, v: V) => void, processDelete: (k: K, v: V) => void, processModify: (k: K, vOld: V, vNew: V) => void): void {
+    // If comparators are different, then the fast path is not possible. Simply fall back to a standard full enumeration.
+    // Also do this if either tree is empty, as everything will be an add/delete and requires a full walk.
+    if (destination._compare !== this._compare || this.isEmpty || destination.isEmpty) {
+      this.forEachPair((kThis, vThis) => {
+        if (!destination.has(kThis)) {
+          processDelete(kThis, vThis);
+        } else {
+          const vDst = destination.get(kThis);
+          if (vThis !== vDst)
```

Comment:
I think `object.is` would be a better comparison to use here, since we implicitly assume all things are equal to themselves (by having the early exits for shared parts of trees) and `!==` violates that for NaN.

Additionally object.is will report all observable differences, even -0 vs +0, which could in some cases cause invalidation a consumer cairs about. '!==' misses that case. If someone does want to to specialized equality (like consider +0 and -0 the same) they can always add that to their processModify handler.